src/awkward/operations/structure.py

```diff
@@ -1460,6 +1460,60 @@ def apply(layout):
         return out
 
 
+def unflatten(array, counts, highlevel=True):
+    """
+    Args:
+        array: Data to create an array with an additional level from.
+        counts (int or array): Number of elements the new level should have.
+        highlevel (bool): If True, return an #ak.Array; otherwise, return
+            a low-level #ak.layout.Content subclass.
+
+    Returns an array with an additional level of nesting. This is roughly the
+    inverse of #ak.flatten, where `counts` were obtained by #ak.num (both with
+    `axis=1`).
+    For example,
```

Comment:
Thanks for including documentation with examples! For the "For example" to be a new paragraph, it just needs a line break separating it from the above.

Maybe the `counts` argument documentation should say that the array should be one-dimensional and may include None values (if you go that route: see my comment below).